store/buildPlayer.js

```diff
@@ -1,121 +1,75 @@
 
 /*
-* Functions to build/cache player object
-* Currently doesn't support search by name
+Functions to build/cache player object
+Currently doesn't support search by name
  */
-function getPlayerData(uuid, cb) {
-  const { url } = generateJob('player', {
-    id: uuid,
-  });
-  getData(redis, url, (err, body) => {
-    if (err) {
-      return cb(err.message, null);
-    }
-    processPlayerData(body.player || {}, (player) => cb(null, player));
-  });
-}
-
-function cachePlayer(player, uuid, key, caching, cb) {
-  if (caching.cacheResult === false) return cb(player);
-  if (config.ENABLE_PLAYER_CACHE) {
-    cacheFunctions.write({
-      key,
-      duration: config.PLAYER_CACHE_SECONDS,
-    }, player);
-  }
-  if (config.ENABLE_DB_CACHE) {
-    queries.insertPlayer(uuid, player, (err) => {
-      if (err) {
-        logger.error(err);
-      }
-    });
-  }
-  return cb(player);
-}
+async function buildPlayer(uuid, { shouldCache = true } = {}) {
+  return cachedFuction(`player:${uuid}`, async () => {
+    const body = await getData(redis, generateJob('player', { id: uuid }).url);
+    const playerData = await pify(processPlayerData)(body.player || {});
 
-function buildPlayer(uuid, cb) {
-  let u;
-  const caching = uuid.caching || {};
-  if (typeof uuid === 'object') {
-    u = uuid.uuid;
-  } else {
-    u = uuid;
-  }
-  const key = `player:${u}`;
-  cacheFunctions.read({ key }, (player) => {
-    if (player) {
-      logger.debug(`Cache hit for player ${u}`);
-      return cb(null, player);
+    if (shouldCache && config.ENABLE_DB_CACHE) {
+      await queries.insertPlayer(uuid, playerData);
     }
-    getPlayerData(u, (err, player) => {
-      if (err) {
-        return cb(err);
-      }
-      cachePlayer(player, u, key, caching, (player) => cb(null, player));
-    });
+
+    return playerData;
+  }, {
+    cacheDuration: config.PLAYER_CACHE_SECONDS,
+    shouldCache: shouldCache && config.ENABLE_PLAYER_CACHE,
   });
 }
 
-function getPlayer(name, cb) {
-  getUUID(name, (err, uuid) => {
-    if (err) {
-      return cb({ status: 404, message: err });
+async function getPlayer(name) {
+  try {
+    const uuid = await getUUID(name);
+    try {
+      return await buildPlayer(uuid);
+    } catch (error) {
+      return { status: 500, message: error.message };
     }
-    buildPlayer(uuid, (err, player) => {
-      if (err) {
-        return cb({ status: 500, message: err });
-      }
-      return cb(null, player);
-    });
-  });
+  } catch (error) {
+    return { status: 404, message: error.message };
+  }
 }
 
-function populatePlayers(players, cb) {
-  async.map(players, (player, done) => {
+async function populatePlayers(players) {
+  return async.map(players, async (player) => {
     const { uuid } = player;
-    queries.getPlayerProfile(uuid, (err, profile, isCached) => {
-      if (err) {
-        logger.error(err);
-      }
+    try {
+      const [profile, isCached] = pify(queries.getPlayerProfile, {
```

Comment:
This seems to return a `pify is not a function` error

Reply:
Is there a stack trace?

Reply:
```
at /usr/src/store/buildPlayer.js:51:35
     at /usr/src/node_modules/async/dist/async.js:148:38
     at /usr/src/node_modules/async/dist/async.js:243:13
     at eachOfArrayLike (/usr/src/node_modules/async/dist/async.js:500:13)
     at eachOf (/usr/src/node_modules/async/dist/async.js:551:16)
     at awaitable (/usr/src/node_modules/async/dist/async.js:208:32)
     at _asyncMap (/usr/src/node_modules/async/dist/async.js:241:16)
    at Object.map (/usr/src/node_modules/async/dist/async.js:594:16)
     at /usr/src/node_modules/async/dist/async.js:216:25
     at new Promise (<anonymous>)
     at Object.awaitable [as map] (/usr/src/node_modules/async/dist/async.js:211:20)
     at populatePlayers (/usr/src/store/buildPlayer.js:48:16)
|     at func (/usr/src/routes/spec.js:821:39)
     at processTicksAndRejections (internal/process/task_queues.js:97:5)

```

Reply:
There was a missing `await` which caused the values to be extracted from the Promise object instead of the result.